C compile error with --gc:arc on version 1.2.0 "unknown type name 'TGenericSeq'

When compiling with --gc:arc, I get a C compilation error.

Before minimizing this error, I also saw nim compilation error "Error: system module needs: newSeq"

Command to repro:  `nim c --gc:arc repro.nim`

### Minimal Example
```nim
import marshal
import streams
let s = newStringStream("")
store(s, "abc")
s.setPosition(0)
echo s.readAll()
```

### Current Output
```
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c: In function ‘len__jGHBAUCABscGrmbUZrbifg’:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:427:3: error: unknown type name ‘TGenericSeq’; did you mean ‘GenericSeqSize’?
  427 |   TGenericSeq* pgenSeq;
      |   ^~~~~~~~~~~
      |   GenericSeqSize
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:430:15: error: ‘TGenericSeq’ undeclared (first use in this function); did you mean ‘GenericSeqSize’?
  430 |   pgenSeq = ((TGenericSeq*) ((*((void**) (x.value)))));
      |               ^~~~~~~~~~~
      |               GenericSeqSize
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:430:15: note: each undeclared identifier is reported only once for each function it appears in
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:430:27: error: expected expression before ‘)’ token
  430 |   pgenSeq = ((TGenericSeq*) ((*((void**) (x.value)))));
      |                           ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:444:23: error: request for member ‘len’ in something not a structure or union
  444 |    result = (*pgenSeq).len;
      |                       ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c: In function ‘X5BX5D___wz9aB9bjTou5XSSHmpDyUEcg’:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:692:61: error: ‘TGenericSeq’ undeclared (first use in this function); did you mean ‘GenericSeqSize’?
  692 |    T25_ = lteqpercent___hPljn3JCDQ00ua1R07X9bxQ_2system((*((TGenericSeq*) (s))).len, i);
      |                                                             ^~~~~~~~~~~
      |                                                             GenericSeqSize
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:692:73: error: expected expression before ‘)’ token
  692 |    T25_ = lteqpercent___hPljn3JCDQ00ua1R07X9bxQ_2system((*((TGenericSeq*) (s))).len, i);
      |                                                                         ^
In file included from /home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:9:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:706:35: error: expected expression before ‘)’ token
  706 |     if (nimSubInt((*((TGenericSeq*) (s))).len, ((NI) 1), &TM__dN9bl2iMoBvVqa2XSK9aEu2w_24)) { raiseOverflow(); goto LA1_;
      |                                   ^
/home/qbradley/.choosenim/toolchains/nim-1.2.0/lib/nimbase.h:568:60: note: in definition of macro ‘nimSubInt’
  568 |     #define nimSubInt(a, b, res) __builtin_ssubll_overflow(a, b, (long long int*)res)
      |                                                            ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:734:35: error: expected expression before ‘)’ token
  734 |     if (nimSubInt((*((TGenericSeq*) (s))).len, ((NI) 1), &TM__dN9bl2iMoBvVqa2XSK9aEu2w_28)) { raiseOverflow(); goto LA1_;
      |                                   ^
/home/qbradley/.choosenim/toolchains/nim-1.2.0/lib/nimbase.h:568:60: note: in definition of macro ‘nimSubInt’
  568 |     #define nimSubInt(a, b, res) __builtin_ssubll_overflow(a, b, (long long int*)res)
      |                                                            ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c: In function ‘getDiscriminant__y0M7U0oBJMrUTHJyPkuAiQ’:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:869:15: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘kind’
  869 |   if (!!(((*n).kind == ((tyEnum_TNimNodeKind__zzKO4S1OVotphEKLvN9bFPQ) 3)))) goto LA3_;
      |               ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:880:16: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘typ’
  880 |  switch ((*(*n).typ).size) {
      |                ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:887:61: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘offset’
  887 |   T6_ = pluspercent___dgYAo7RfdUVVpvkfKDym8w_2system(a, (*n).offset);
      |                                                             ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:897:61: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘offset’
  897 |   T8_ = pluspercent___dgYAo7RfdUVVpvkfKDym8w_2system(a, (*n).offset);
      |                                                             ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:907:62: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘offset’
  907 |   T10_ = pluspercent___dgYAo7RfdUVVpvkfKDym8w_2system(a, (*n).offset);
      |                                                              ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:917:62: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘offset’
  917 |   T12_ = pluspercent___dgYAo7RfdUVVpvkfKDym8w_2system(a, (*n).offset);
      |                                                              ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c: In function ‘selectBranch__2QflL29bYJlQDZ49b8lLF73A’:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:972:62: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
  972 |   T3_ = ltpercent___hPljn3JCDQ00ua1R07X9bxQsystem(discr, (*n).len);
      |                                                              ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:978:16: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘sons’
  978 |   result = (*n).sons[(discr)- 0];
      |                ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:983:17: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
  983 |    if ((NU)((*n).len) > (NU)(32766)){ raiseIndexError2((*n).len, 32766); goto BeforeRet_;
      |                 ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:983:60: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
  983 |    if ((NU)((*n).len) > (NU)(32766)){ raiseIndexError2((*n).len, 32766); goto BeforeRet_;
      |                                                            ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:985:17: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘sons’
  985 |    result = (*n).sons[((*n).len)- 0];
      |                 ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:985:28: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
  985 |    result = (*n).sons[((*n).len)- 0];
      |                            ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:994:16: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
  994 |   if ((NU)((*n).len) > (NU)(32766)){ raiseIndexError2((*n).len, 32766); goto BeforeRet_;
      |                ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:994:59: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
  994 |   if ((NU)((*n).len) > (NU)(32766)){ raiseIndexError2((*n).len, 32766); goto BeforeRet_;
      |                                                           ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:996:16: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘sons’
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:996:16: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘sons’
  996 |   result = (*n).sons[((*n).len)- 0];
      |                ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:996:27: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
  996 |   result = (*n).sons[((*n).len)- 0];
      |                           ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c: In function ‘fieldsAux__3aptd1t9avVoXDW89bnYBhxw’:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1014:14: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘kind’
 1014 |  switch ((*n).kind) {
      |              ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1034:20: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘name’
 1034 |   colontmpD_ = (*n).name;
      |                    ^
In file included from /home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:9:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1038:45: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘offset’
 1038 |   if (nimAddInt(((NI) (ptrdiff_t) (p)), (*n).offset, &TM__dN9bl2iMoBvVqa2XSK9aEu2w_36)) { raiseOverflow(); goto BeforeRet_;
      |                                             ^
/home/qbradley/.choosenim/toolchains/nim-1.2.0/lib/nimbase.h:567:63: note: in definition of macro ‘nimAddInt’
  567 |     #define nimAddInt(a, b, res) __builtin_saddll_overflow(a, b, (long long int*)res)
      |                                                               ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1043:103: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘typ’
 1043 |   T8_ = newAny__6x8Rdn7KRQqWLTQb2KSnAgtypeinfo(((void*) ((NI)(TM__dN9bl2iMoBvVqa2XSK9aEu2w_36))), (*n).typ);
      |                                                                                                       ^
In file included from /home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:9:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1078:22: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
 1078 |    if (nimSubInt((*n).len, ((NI) 1), &TM__dN9bl2iMoBvVqa2XSK9aEu2w_40)) { raiseOverflow(); goto BeforeRet_;
      |                      ^
/home/qbradley/.choosenim/toolchains/nim-1.2.0/lib/nimbase.h:568:60: note: in definition of macro ‘nimSubInt’
  568 |     #define nimSubInt(a, b, res) __builtin_ssubll_overflow(a, b, (long long int*)res)
      |                                                            ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1095:49: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘sons’
 1095 |      fieldsAux__3aptd1t9avVoXDW89bnYBhxw(p, (*n).sons[(i)- 0], ret);
      |                                                 ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1119:22: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘name’
 1119 |   colontmpD__2 = (*n).name;
      |                      ^
In file included from /home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:9:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1123:45: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘offset’
 1123 |   if (nimAddInt(((NI) (ptrdiff_t) (p)), (*n).offset, &TM__dN9bl2iMoBvVqa2XSK9aEu2w_46)) { raiseOverflow(); goto BeforeRet_;
      |                                             ^
/home/qbradley/.choosenim/toolchains/nim-1.2.0/lib/nimbase.h:567:63: note: in definition of macro ‘nimAddInt’
  567 |     #define nimAddInt(a, b, res) __builtin_saddll_overflow(a, b, (long long int*)res)
      |                                                               ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1128:104: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘typ’
 1128 |   T21_ = newAny__6x8Rdn7KRQqWLTQb2KSnAgtypeinfo(((void*) ((NI)(TM__dN9bl2iMoBvVqa2XSK9aEu2w_46))), (*n).typ);
      |                                                                                                        ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c: In function ‘getEnumField__gVv9aCkk9agum2FqwlrKdc2g’:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1204:70: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
 1204 |    T11_ = ltpercent___hPljn3JCDQ00ua1R07X9bxQsystem(e, (*(*typ).node).len);
      |                                                                      ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1210:42: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘sons’
 1210 |    result = cstrToNimstr((*(*(*typ).node).sons[(e)- 0]).name);
      |                                          ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1225:11: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘sons’
 1225 |   s = (*n).sons;
      |           ^
In file included from /home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:9:
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1235:22: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘len’
 1235 |    if (nimSubInt((*n).len, ((NI) 1), &TM__dN9bl2iMoBvVqa2XSK9aEu2w_51)) { raiseOverflow(); goto BeforeRet_;
      |                      ^
/home/qbradley/.choosenim/toolchains/nim-1.2.0/lib/nimbase.h:568:60: note: in definition of macro ‘nimSubInt’
  568 |     #define nimSubInt(a, b, res) __builtin_ssubll_overflow(a, b, (long long int*)res)
      |                                                            ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1255:25: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘offset’
 1255 |       if (!((*s[(i)- 0]).offset == e)) goto LA20_;
      |                         ^
/home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c:1258:41: error: ‘TNimNode’ {aka ‘struct TNimNode’} has no member named ‘name’
 1258 |       result = cstrToNimstr((*s[(i)- 0]).name);
      |                                         ^
Error: execution of an external compiler program 'gcc -c  -w   -I/home/qbradley/.choosenim/toolchains/nim-1.2.0/lib -I/home/qbradley -o /home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c.o /home/qbradley/.cache/nim/repro_d/stdlib_typeinfo.nim.c' failed with exit code: 1
```

### Expected Output
```
"abc"
```

### Additional Information

* It only fails with --gc:arc.  Remove that command line option and it compiles and produces the correct result.
* commenting out line 4 `store(s, "abc")` and it compiles even with --gc:arc

```
$ nim -v
Nim Compiler Version 1.2.0 [Linux: amd64]
Compiled at 2020-04-03
Copyright (c) 2006-2020 by Andreas Rumpf

active boot switches: -d:release
```

--- a/lib/pure/marshal.nim
+++ b/lib/pure/marshal.nim
@@ -50,6 +50,11 @@
 ## * `streams module <streams.html>`_
 ## * `json module <json.html>`_
 
+when defined(nimV2):
+  {.error: """marshal module is not supported in new runtime.
+Please use alternative packages for serialization. 
+It is possible to reimplement this module using generics and type traits. 
+Please contribute new implementation.""".}
 
 import streams, typeinfo, json, intsets, tables, unicode
 

